src/app/game.ts

```diff
@@ -448,6 +449,108 @@ export class Game {
     this.storageCapacity += parseInt((warehouse.capacity * 1.5).toFixed(0))
   }
 
+  private checkIfRandomEvent() {
+    const randomInt = Math.floor(Math.random() * 10000)
+
+    if (randomInt < 30) {
+      this.handleRandomEvent()
+    } else {
+      return
+    }
+  }
+
+  private handleRandomEvent() {
+    const randomInt = Math.floor(Math.random() * 5)
+
+    switch (randomInt) {
+      case 0:
+        this.handleRandomGoldStealEvent()
+        break
+      case 1:
+        this.handleRandomBuilidngLevelDecreaseEvent()
+        break
+      case 2:
+        this.handleRandomWoodIncreaseEvent()
+        break
+      case 3:
+        this.handleMurderEvent()
+        break
+      case 4:
+        this.handleBornEvent()
+      default:
+        break
+    }
+  }
+
+  private handleRandomGoldStealEvent() {
+    if (this.getGoldAmount() === 0) {
+      return
+    }
+    const goldTaken = Math.floor(0.05 * this.getGoldAmount())
+    const gold = Math.floor(0.95 * this.getGoldAmount())
+    this.changeGoldAmount(gold)
+    this.addGameMessage({
+      message: `Oh no, thieves have stolen ${goldTaken} gold from your storage`,
+      type: MessageType.ERROR,
+    })
+  }
+
+  private handleRandomBuilidngLevelDecreaseEvent() {
+    const randomBuilding = this.buildings[
+      Math.floor(Math.random() * this.buildings.length - 1)
+    ]
+
+    if (randomBuilding.level === 0) {
+      return
+    }
+
+    randomBuilding.level = randomBuilding.getLevel() - 1
```

Comment:
we set somewhere level of the building outside building class? If no then level can be private and you should use getLevel() function, otherwise we can delete getLevel() function and directly use level property.